pkg/gatewayserver/io/io.go

```diff
@@ -542,9 +545,46 @@ func (c *Connection) ConnectTime() time.Time { return time.Unix(0, c.connectTime
 
 // StatusStats returns the status statistics.
 func (c *Connection) StatusStats() (last *ttnpb.GatewayStatus, t time.Time, ok bool) {
```

Comment:
The return value of `StatusStats()` should not be changed by any callers.

I think the solution is not copying the status on get, but making sure that an immutable instance is set.

Reply:
Does that not mean that anyone who updates the gateway status (and by extension, any of the connection stats) should make sure (now and in the future) should be careful to make the change immutable? How do we enforce this?

Also, what would be a good way to set an immutable instance of the gateway status?

Reply:
Updated code to store a copy of the gateway stats instead.